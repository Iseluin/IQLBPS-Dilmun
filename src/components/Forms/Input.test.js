import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react'
import { Input } from '.'

test('Input renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should have a value', () => {
    render(<Input value="test" />);
    const value = screen.getByDisplayValue(/test/i)
    expect(value).toBeTruthy();
})

test('should have a disabled', () => {
    render(<Input disabled />);
    const input = screen.getByRole('textbox')
    expect(input).toBeDisabled();
})

test('should have a error', () => {
    render(<Input errors />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveClass('error');
})

test('should have a className', () => {
    render(<Input className="test" />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveClass('test');
})

test('should have a style', () => {
    render(<Input style={{ color: 'red' }} />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveStyle('color: red');
})

test('should have a id', () => {
    render(<Input id="test" />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveAttribute('id', 'test');
})

test('should have a type', () => {
    render(<Input type="number" />);
    const input = screen.getByRole('spinbutton')
    expect(input).toHaveAttribute('type', 'number');
})

test('should have a placeholder', () => {
    render(<Input placeholder="test" />);
    const input = screen.getByPlaceholderText(/test/i)
    expect(input).toBeTruthy();
})

test('should have a label', () => {
    render(<Input>test</Input>);
    const input = screen.getByText(/test/i)
    expect(input).toBeTruthy();
})
