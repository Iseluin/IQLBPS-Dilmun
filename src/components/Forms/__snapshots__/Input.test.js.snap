// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="relative border-2 rounded-md mb-2 text-background border-tertiary "
>
  <input
    className="outline-none w-full rounded bg-transparent text-sm transition-all duration-200 ease-in-out p-2  "
    disabled={false}
    onClick={[Function]}
  />
  <label
    className="absolute top-0 left-0 flex items-center text-tertiary text-opacity-50 p-2 transition-all duration-200 ease-in-out text-sm "
  />
</div>
`;
